File: java/src/test/java/implementation/TargetTest.java
package implementation;

import org.json.JSONObject;
import org.junit.Before;
import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.assertEquals;

public class TargetTest {

    private Wrapper wrapper;
    private JSONObject target = new JSONObject("{\n" +
            "'civilité': 'Mr',\n" +
            "8630: 'Doe',\n" +
            "8628: 'test@test.com'\n" +
            "}");
    private String segmentId = "14091";
    private String actionId = "000QF0";

    @Before
    public void setUp() {
        JSONObject config = new JSONObject();

        config.put("xkey", "my_xkey");
        config.put("url", "http://127.0.0.1:8888/");
        wrapper = new Wrapper(config);
    }

    @Test
    public void testCreateOrModifyTarget() {
        JSONObject response = this.wrapper.createOrModifyTarget(this.target);
        assertEquals(response.getInt("statusCode"), 200);
    }

    @Test
    public void testCreateTargetAndAddToSegment() {
        JSONObject response = this.wrapper.createTargetAndAddToSegment(this.target, this.segmentId);
        assertEquals(response.getInt("statusCode"), 204);
    }

    @Test
    public void testAddTargetAndSendMessage() {
        JSONObject response = this.wrapper.addTargetAndSendMessage(target, actionId);
        assertEquals(response.getInt("statusCode"), 204);
    }
}

File: csharp/DriverTest/Implementation/TargetTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;
using RestSharp;
using NP6Api.Implementation;
using System.Net;

namespace DriverTest.Implementation
{
    [TestClass]
    public class TargetTest
    {
        private static Wrapper _wrapper;
        private static JsonObject _target;
        private static string _actionId;
        private static string _segmentId;

        [ClassInitialize]
        public static void CampaignTestInitialize(TestContext testContext)
        {
            var config = new Dictionary<string, string>();

            config["xKey"] = "YOUR XKEY";
            config["url"] = "http://127.0.0.1:8888/";
            _wrapper = new Wrapper(config);
            _target = new JsonObject();
            _target.Add("civilité", "Mr");
            _target.Add("1234", "Doe");
            _target.Add("1234", "test@test.com");
            _actionId = "000ABC";
            _segmentId = "12345";
        }

        [TestMethod]
        public void TestCreateOrModifyTarget()
        {
            var response = _wrapper.CreateOrModifyTarget(_target);

            Assert.AreEqual((HttpStatusCode)200, response.StatusCode);
        }

        [TestMethod]
        public void TestAddTargetAndSendMessage()
        {
            var response = _wrapper.AddTargetAndSendMessage(_target, _actionId);

            Assert.AreEqual((HttpStatusCode)204, response.StatusCode);
        }

        [TestMethod]
        public void TestCreateTargetAndAddToSegment()
        {
            var response = _wrapper.CreateTargetAndAddToSegment(_target, _segmentId);

            Assert.AreEqual((HttpStatusCode)204, response.StatusCode);
        }
    }
}
